include/crocoddyl/multibody/residuals/frame-placement.hxx

```diff
@@ -0,0 +1,88 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2021, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#include <pinocchio/algorithm/frames.hpp>
+#include "crocoddyl/multibody/residuals/frame-placement.hpp"
+
+namespace crocoddyl {
+
+template <typename Scalar>
+ResidualModelFramePlacementTpl<Scalar>::ResidualModelFramePlacementTpl(boost::shared_ptr<StateMultibody> state,
+                                                                       const pinocchio::FrameIndex id, const SE3& pref,
+                                                                       const std::size_t nu)
+    : Base(state, 6, nu), id_(id), pref_(pref), oMf_inv_(pref.inverse()), pin_model_(state->get_pinocchio()) {
+  v_dependent_ = false;
+  u_dependent_ = false;
+}
+
+template <typename Scalar>
+ResidualModelFramePlacementTpl<Scalar>::ResidualModelFramePlacementTpl(boost::shared_ptr<StateMultibody> state,
+                                                                       const pinocchio::FrameIndex id, const SE3& pref)
+    : Base(state, 6), id_(id), pref_(pref), oMf_inv_(pref.inverse()), pin_model_(state->get_pinocchio()) {
+  v_dependent_ = false;
+  u_dependent_ = false;
+}
+
+template <typename Scalar>
+ResidualModelFramePlacementTpl<Scalar>::~ResidualModelFramePlacementTpl() {}
+
+template <typename Scalar>
+void ResidualModelFramePlacementTpl<Scalar>::calc(const boost::shared_ptr<ResidualDataAbstract>& data,
+                                                  const Eigen::Ref<const VectorXs>&,
+                                                  const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+
+  // Compute the frame placement w.r.t. the reference frame
+  pinocchio::updateFramePlacement(*pin_model_.get(), *d->pinocchio, id_);
+  d->rMf = oMf_inv_ * d->pinocchio->oMf[id_];
+  d->r = pinocchio::log6(d->rMf);
+  data->r = d->r;  // this is needed because we overwrite it
```

Comment:
Why do we need to overwrite it? Sorry I don't understand.

Reply:
Interesting ... I see where was my mistake. We need to write
```cpp
data->r = pinocchio::log6(d->rMf).toVector();
```
instead of
```cpp
data->r = pinocchio::log6(d->rMf);
```

If we do not use `toVector()`, then Pinocchio expects a vector with fixed dimension.

**In short, I have adapted both cases: frame-placement and frame-rotation. Thanks for the tip!**